test/extensions/http/header_validators/envoy_default/http1_header_validator_test.cc

```diff
@@ -0,0 +1,498 @@
+#include "source/extensions/http/header_validators/envoy_default/http1_header_validator.h"
+
+#include "test/extensions/http/header_validators/envoy_default/header_validator_test.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Http {
+namespace HeaderValidators {
+namespace EnvoyDefault {
+namespace {
+
+using ::Envoy::Http::HeaderString;
+using ::Envoy::Http::Protocol;
+
+class Http1HeaderValidatorTest : public HeaderValidatorTest {
+protected:
+  Http1HeaderValidatorPtr createH1(absl::string_view config_yaml) {
+    envoy::extensions::http::header_validators::envoy_default::v3::HeaderValidatorConfig
+        typed_config;
+    TestUtility::loadFromYaml(std::string(config_yaml), typed_config);
+
+    return std::make_unique<Http1HeaderValidator>(typed_config, Protocol::Http11, stream_info_);
+  }
+};
+
+TEST_F(Http1HeaderValidatorTest, ValidateTransferEncoding) {
+  HeaderString valid{"gzip, chunked"};
+  HeaderString invalid{"{deflate}"};
+  auto uhv = createH1(empty_config);
+
+  EXPECT_TRUE(uhv->validateTransferEncodingHeader(valid).ok());
+
+  EXPECT_REJECT_WITH_DETAILS(uhv->validateTransferEncodingHeader(invalid),
+                             "uhv.http1.invalid_transfer_encoding");
+}
+
+TEST_F(Http1HeaderValidatorTest, ValidateRequestHeaderEntryEmpty) {
+  HeaderString empty{""};
+  HeaderString value{"foo"};
+  auto uhv = createH1(empty_config);
+
+  EXPECT_REJECT_WITH_DETAILS(uhv->validateRequestHeaderEntry(empty, value),
+                             UhvResponseCodeDetail::get().EmptyHeaderName);
+}
+
+TEST_F(Http1HeaderValidatorTest, ValidateRequestHeaderEntryMethod) {
+  HeaderString name{":method"};
+  HeaderString valid{"GET"};
+  HeaderString invalid{"CUSTOM-METHOD"};
```

Comment:
I find it confusing that an "invalid" method value would be accepted by the validator.  Consider adding a comment, or maybe renaming `valid` to `defined` and `invalid` to `custom` or similar.